src/DisputeHandler.ts

```diff
@@ -55,164 +72,281 @@ class DisputeHandler {
-        conflictingBlocks: SignedBlockStruct[]
-    ): Promise<void> {
-        const proof =
-            this.proofManager.createDoubleSignProof(conflictingBlocks);
-        const _firstBlock = EvmUtils.decodeBlock(
-            conflictingBlocks[0].encodedBlock
-        );
-        return this.createDispute(
-            _firstBlock.transaction.header.forkCnt,
-            NO_PARTICIPANT_TO_FOLD,
-            INITIAL_TRANSACTION_COUNT,
-            [proof]
-        );
-    }
 
-    public async disputeIncorrectData(
-        incorrectBlockSigned: SignedBlockStruct
-    ): Promise<void> {
-        const proof =
-            this.proofManager.createIncorrectDataProof(incorrectBlockSigned);
-        const _block = EvmUtils.decodeBlock(incorrectBlockSigned.encodedBlock);
-        return this.createDispute(
-            _block.transaction.header.forkCnt,
-            NO_PARTICIPANT_TO_FOLD,
-            INITIAL_TRANSACTION_COUNT,
-            [proof]
-        );
-    }
-
-    // Not needed publicly - just internaly
-    // public async disputeNewerState(
-    //     forkCnt: number,
-    //     participantAdr: AddressLike
-    // ): Promise<void> {
-    //     let proof = this.createNewerStateProof(forkCnt, participantAdr, );
-    //     if (!proof) return;
-    //     await this.createDispute(forkCnt, participantAdr, 0, [proof]);
-    // }
+    public addDispute(
+        forkCnt: number,
+        dispute: DisputeStruct,
+        disputeAuditingData: DisputeAuditingDataStruct
+    ): void {}
 
-    public async disputeFoldPriorBlock(
-        forkCnt: BigNumberish,
-        transactionCnt: number
-    ): Promise<void> {
-        const proof = ProofManager.createFoldPriorBlockProof(transactionCnt);
-        return this.createDispute(
-            forkCnt,
-            NO_PARTICIPANT_TO_FOLD,
-            INITIAL_TRANSACTION_COUNT,
-            [proof]
-        );
+    // creating a dispute auditing data struct
+    public createDisputeAuditingData(
+        forkCnt: number,
+        transactionCnt: number,
+        outputStateSnapshot: StateSnapshotStruct,
+        timestamp: BigNumberish,
+        prevDispute: DisputeStruct,
+        prevDisputeTimestamp: BigNumberish
+    ): DisputeAuditingDataStruct {
+        // TODO: Implement proper dispute auditing data creation
+        const latestStateStateMachineState =
+            this.agreementManager.getStateMachineState(
+                forkCnt,
+                transactionCnt
+            )!;
+        return {
+            genesisStateSnapshot: outputStateSnapshot,
+            latestStateSnapshot: this.agreementManager.getSnapShot(
+                forkCnt,
+                transactionCnt
+            )!,
+            outputStateSnapshot: outputStateSnapshot,
+            milestoneSnapshots: [],
+            latestStateStateMachineState: latestStateStateMachineState,
+            joinChannelBlocks: this.agreementManager.getJoinChannelChain(),
+            timestamp: 0,
```

Comment:
Timestamp shouldn't be part of auditing data since you don't know it at creation - it's set on-chain so it's part of the commitment and stored separately and uploaded separately when needed to verify the commitment/used